editor.h

```diff
@@ -3,20 +3,91 @@
 
 #include <gtkmm.h>
 
+#include <unordered_set>
+
+#include "geometry.h"
+#include "map.h"
+
+enum Tool {
+  SELECT,
+  CIRCLE,
+};
+
+class ToolState {
+ public:
+  virtual void click(Point p) = 0;
+  Tool type();
+  virtual ~ToolState() = default;
+
+ protected:
+  ToolState(Tool t);
+  Tool t;
+};
+
+class SelectState : public ToolState {
+ public:
+  SelectState();
+  void click(Point p);
+};
+
+class CircleState : public ToolState {
+ public:
+  CircleState();
+  void click(Point p);
+};
+
+class MapView {
+ public:
+  MapView();
+
+  Point map_to_screen(Point p, double width, double height);
```

Comment:
nit: Valószínűleg érdemes stock libraryt használni. Én pl. az Euler3-at tartom elég jónak erre. Addig is talán érdemesebb Vector2f-nek vagy hasonlónak hívni.